src/expr/src/explain/json.rs

```diff
@@ -43,12 +44,33 @@ where
         let sources = self
             .sources
             .iter()
-            .map(|ExplainSource { id, op, .. }| {
-                serde_json::json!({
-                    "id": id,
-                    "op": op
-                })
-            })
+            .map(
+                |ExplainSource {
+                     id,
+                     op,
+                     pushdown_info,
+                 }| {
+                    let mut json = serde_json::json!({
+                        "id": id,
+                        "op": op,
+                    });
+
+                    if let Some(PushdownInfo { trace }) = pushdown_info {
```

Comment:
Why are you not serializing the entire `PushdownInfo` here but only a derived boolean for each `trace` entry?

Reply:
The idea was: right now we track a ternary flag for each column: definitely pushdownable, definitely not pushdownable, or "maybe" (if a relevant function is not annotated). This third state is probably temporary; in particular, I hope to turn `monotonic` into a function annotation where a boolean is probably more appropriate. Right now it's useful to express that uncertainty in the human-facing output, but if we're writing tests against the machine-parsable output it seems less appropriate to rely on it.

However, don't feel strongly; if you have another preference I'm happy to switch!

Reply:
> However, don't feel strongly; if you have another preference I'm happy to switch!

I think it's OK to have this as a transient state, the only strong opinion that I have is to not leave things unaligned for too long.

Reply:
Okay, made a note to address this in the followup!